gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java

```diff
@@ -264,23 +386,42 @@ private int length(GroovySourceToken token) {
         return offset2 - offset1;
     }
 
+    private String getStyleClass(Token token2) {
+        String style = "";
+        style = !bracketsMatches.searchingForMatches ? styleClass(token2.getType()) : bracketMatchesStyleClass(token2.getType());
+        return style;
+    }
+
+    private void addPostion(int postion) {
+        if (!bracketsMatches.searchingForMatches) {
+            codeArea.tokensPositions.add(postion);
+        }
+    }
+
     private StyleSpans<Collection<String>> computeHighlighting(String text) {
+        String textUsed = text;
+        codeArea.tokensPositions.clear();
+        if (!codeArea.getText().isEmpty() && bracketsMatches.searchingForMatches) {
+            textUsed = codeArea.getText(bracketsMatches.matchedPosition, bracketsMatches.matchedPosition + 1);
+        }
         Stopwatch stopwatch = Stopwatch.createStarted();
 
         boolean added = false;
         StyleSpansBuilder<Collection<String>> spansBuilder = new StyleSpansBuilder<>();
-        if (!text.isEmpty()) {
+        if (!textUsed.isEmpty()) {
             SourceBuffer sourceBuffer = new SourceBuffer();
-            try (UnicodeEscapingReader reader = new UnicodeEscapingReader(new StringReader(text), sourceBuffer)) {
+            try (UnicodeEscapingReader reader = new UnicodeEscapingReader(new StringReader(textUsed), sourceBuffer)) {
                 GroovyLexer lexer = new GroovyLexer(new UnicodeLexerSharedInputState(reader));
                 lexer.setWhitespaceIncluded(true);
                 TokenStream tokenStream = lexer.plumb();
                 Token token = tokenStream.nextToken();
                 while (token.getType() != Token.EOF_TYPE) {
-                    String styleClass = styleClass(token.getType());
+                    String styleClass = getStyleClass(token);
                     int length = length((GroovySourceToken) token);
                     if (styleClass != null) {
                         spansBuilder.add(Collections.singleton(styleClass), length);
+                        int tokenPostion = codeArea.getDocument().position(token.getLine() - 1, token.getColumn() - 1).toOffset();
+                        addPostion(tokenPostion);
```

Comment:
It seems you are indexing all the code? Is it really efficient to do like this. Isn’t it more efficient to look around the carret to find a match?